### PR TITLE
Prepared documentation site to host multiple versions and external components

### PR DESCRIPTION
**Pull request by lefou**
This pull request is my attempt to improve the following aspects of our current documentation site:

* Better maintainability (by switching to AsciiDoc/Asciidoctor, which e.g. supports variables to easily update all references to the current version at once)
* Easier contribution (each page has a direct edit-link in the upper right corder)
* Multi-Version documentation (by using Antora as site generator)
* Potential to embed more external documentation, e.g. for plugins (by using Antora as site generator)

I create a gh-pages branch in my personal fork, which you can use as preview.

https://lefou.github.io/mill

This PR isn't ready for merge, but for review.

Links: 
* https://asciidoctor.org/
* https://antora.org/

### CODE REVIEW

**Comment by sake92** [comment]
Looks very cool!  
I like the branching solution to version very much. 😄 👍🏻 

Two things we could improve IMHO:
- a bit smaller font size
- use PrismJS instead of HighlightJS
  Then we'd get *really copy-pasteable* CMDs and pretty highlighting  
  You can see example at [my blog:](https://blog.sake.ba/posts/programiranje/java/types.html)   
  When you copy it, you don't get the output lines or bash prompt! :)

**Comment by lefou** [comment]
Ok, I think the page is ready for review. I need to automate the publishing tough.

File: docs/antora/modules/ROOT/pages/Intro_to_Mill.adoc [hunk 1]
```diff
@@ -1,110 +1,124 @@
...
 
 To get started, download Mill and install it into your system via the following
 `curl`/`chmod` command:
 
-```bash
-sudo curl -L https://github.com/com-lihaoyi/mill/releases/download/0.9.5/0.9.5 > /usr/local/bin/mill && sudo chmod +x /usr/local/bin/mill
-```
+[source,bash]
+----
+sudo curl -L {mill-github-url}/releases/download/{mill-last-tag}/{mill-last-tag} > /usr/local/bin/mill && sudo chmod +x /usr/local/bin/mill
```
Comment (sake92) [inline docs/antora/modules/ROOT/pages/Intro_to_Mill.adoc:101]: Seems like these `{mill-github-url}` aren't replaced?
https://lefou.github.io/mill/mill/Intro_to_Mill.html#_manual

File: docs/antora/modules/ROOT/pages/Intro_to_Mill.adoc [hunk 2]
```diff
@@ -847,102 +884,107 @@ publish
...
 `mill-*`. The most useful is `mill-profile.json`, which logs the tasks run and time taken for the last Mill command you
 executed. This is very useful if you want to find out exactly what tasks are being run and Mill is being slow.
 
 Each folder currently contains the following files:
 
-- `dest/`: a path for the `Task` to use either as a scratch space, or to place generated files that are returned
-  using `PathRef`s. `Task`s should only output files within their given `dest/` folder (available as `T.dest`) to avoid
-  conflicting with other `Task`s, but files within `dest/` can be named arbitrarily.
+* `dest/`: a path for the `Task` to use either as a scratch space, or to place generated files that are returned
+ using `PathRef`s. `Task`s should only output files within their given `dest/` folder (available as `T.dest`) to avoid
```
Comment (sake92) [inline docs/antora/modules/ROOT/pages/Intro_to_Mill.adoc:912]: Not sure why, but this looks weird: https://lefou.github.io/mill/mill/Intro_to_Mill.html#_structure_of_the_out_folder
Whole "PathRef`s. `Task`s should only output files within their given `dest" is highlighted as code

File: docs/antora/modules/ROOT/pages/Intro_to_Mill.adoc [hunk 3]
```diff
@@ -678,47 +710,50 @@ mill clean "foo.{compile,run}"
...
 repositories. Note that it uses heuristics based on common versionning schemes, so it may not work as expected for
 dependencies with particularly weird version numbers.
 
 Current limitations:
 
-- Only works for `JavaModule`s (including `ScalaModule`s,
-  `CrossScalaModule`s, etc.) and Maven repositories.
-- Always applies to all modules in the build.
-- Doesn't apply to `$ivy` dependencies used in the build definition itself.
+* Only works for `JavaModule`s (including `ScalaModule`s,
```
Comment (sake92) [inline docs/antora/modules/ROOT/pages/Intro_to_Mill.adoc:728]: These need to be double backticks: "``JavaModule``s" etc.  
 https://docs.antora.org/antora/2.3/asciidoc/monospace/#inline-monospace-syntax

File: docs/antora/modules/ROOT/pages/Configuring_Mill.adoc [hunk 4]
```diff
@@ -285,43 +308,48 @@ object foo extends CommonModule
...
 how you start mill, one of two files will be loaded. For the build REPL
 (`--repl` or `-i` without specifying a target), `~/.mill/ammonite/predef.sc`
 will be loaded, and for builds from the command line the file
 `~/.mill/ammonite/predefScript.sc` will be included. You might want to create
 a symlink from one to the other to avoid duplication.
 
 Example `~/.mill/ammonite/predef.sc`
-```scala
+
+.`~/.mill/ammonite/prefef.sc`
```
Comment (sake92) [inline docs/antora/modules/ROOT/pages/Configuring_Mill.adoc:331]: predef.sc

File: docs/antora/modules/ROOT/pages/Intro_to_Mill.adoc (hunk 3 above)
Comment (sake92) [reply docs/antora/modules/ROOT/pages/Intro_to_Mill.adoc:728]: These are usually plurals, just Ctrl+F for `s 😄

**Comment by lefou** [comment]
@sake92 Thanks for the feedback!

**Pull request merged**